src/rgw/driver/rados/rgw_rest_realm.cc

```diff
@@ -342,11 +363,7 @@ class RGWOp_Realm_List : public RGWRESTOp {
 
 void RGWOp_Realm_List::execute(optional_yield y)
 {
-  {
-    // read default realm
-    RGWRealm realm(driver->ctx(), static_cast<rgw::sal::RadosStore*>(driver)->svc()->sysobj);
-    [[maybe_unused]] int ret = realm.read_default_id(this, default_id, y);
-  }
+  default_id = s->penv.cfgstore->read_default_realm_id(this, y, default_id);
```

Comment:
this function returns an integer error code, which we had previously ignored with `[[maybe_unused]] int ret`. don't assign to `default_id` here, which is the output argument of the function